### PR TITLE
feat: create a new command to run extra tutor commands

### PR DESCRIPTION
**Pull request by bra-i-am**
## Description

This PR creates a new command to run extra commands added into a property `DISTRO_EXTRA_COMMANDS` in the `config.yml` file

## Testing instructions
Access to the new [pipeline of Picasso](https://picasso.dedalo.edunext.co/job/PICASSO_UNDER_CONSTRUCTION/), "Build with Parameter" and click in "Build"; here you'll be testing the commands added in the [config.yml from palm/dedalo strain](https://github.com/eduNEXT/ednx-strains/blob/11c05a57bbdc8c27f89cba46623399599f805f42/palma/dedalo/config.yml#L7). Feel free of modifying the commands as you like.

## Additional information
[JIRA ISSUE DS-826](https://edunext.atlassian.net/jira/software/c/projects/DS/boards/40?selectedIssue=DS-826)

### CODE REVIEW

File: tutordistro/distro/extra_commands/infrastructure/tutor_commands.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+"""
+Distro tutor commands functions.
```
Comment (dcoa) [inline tutordistro/distro/extra_commands/infrastructure/tutor_commands.py:2]:
```suggestion
Distro tutor command functions.
```

File: tutordistro/commands/run_extra_commands.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import click
+from tutor import config as tutor_config
+
+from tutordistro.distro.extra_commands.application.commands_runner import CommandsRunner
+from tutordistro.distro.extra_commands.infrastructure.tutor_commands import (
+    TutorCommandManager,
+)
+
+
+@click.command(name="run-extra-commands", help="Run distro extra commands")
```
Comment (dcoa) [inline tutordistro/commands/run_extra_commands.py:16]:
```suggestion
@click.command(name="run-extra-commands", help="Run tutor commands")
```

File: tutordistro/commands/run_extra_commands.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+def run_extra_commands():
+    """
+    This command runs the distro extra commands
```
Comment (dcoa) [inline tutordistro/commands/run_extra_commands.py:19]:
```suggestion
    This command runs tutor commands defined in DISTRO_EXTRA_COMMANDS
```

**Comment by dcoa** [comment]
I tested it and works pretty well, in the middle of this process I found an edge case, running 2 or more commands in  one line, such as 
`tutor plugins install mfe && tutor plugins enable mfe`

I love being able to use it but it also allows me to run different commands like:
`tutor plugins enable mfe && touch extra.py`

File: tutordistro/distro/extra_commands/infrastructure/tutor_commands.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                    raise CommandError(f'Command "{command}" is not a valid Tutor command. Take the official Tutor commands into account https://docs.tutor.edly.io/reference/cli/index.html')
+            
+            process = subprocess.Popen(
+                command,
+                shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, executable='/bin/bash'
+            )
+
+            output, error = process.communicate()
+            # If a command failed
+            if process.returncode != 0:
```
Comment (Alec4r) [inline tutordistro/distro/extra_commands/infrastructure/tutor_commands.py:49]: this is necessary? or just with the except subprocess.SubprocessError is enough to catch it?

**Review by Alec4r — APPROVED** [review]
It looks nice for me/

**Comment by dcoa** [comment]
Right now I'm experiencing this error when try to run the commands I'm using the same environment that my first testing 

```
⚠️  The current environment stored at /home/dcoa/Documents/eduNext/distro/plugin-tutor/distrocommands/env is not up-to-date: it is at vv16.1.3 while the 'tutor' binary is at v16.1.3. You should upgrade the environment by running:

    tutor config save
```

I'm getting the same error in Picasso https://picasso.dedalo.edunext.co/job/PICASSO_UNDER_CONSTRUCTION/246/console#:~:text=%2B%20tutor%20distro%20run%2Dextra%2Dcommands

I said it's an error because after that any command is running.

**Review by dcoa — APPROVED** [review]
that fix the issue, thanks @bra-i-am

File: tutordistro/distro/extra_commands/infrastructure/tutor_commands.py (hunk 4 above)
Comment (bra-i-am) [reply tutordistro/distro/extra_commands/infrastructure/tutor_commands.py]: @Alec4r thank you for reviewing this, I was using this to print prettier the console output but after some research I saw I could use the subprocess methods in a better way; I already made the change to manage the error only from and except and keep the behaviour just as I wanted :D

**Review by MaferMazu — COMMENTED** [review]
I'm sorry for not getting back to you sooner.

Thanks a lot for this PR; kudos for the test and the structure you built.

The primary function works as expected, but I get an error when I try to run the command run-extra-commands without defining DISTRO_EXTRA_COMMANDS. That behavior will generate errors in Picasso because we will run extra commands as part of the pipeline, and if someone doesn't need extra commands and doesn't have the variable, this breaks.

Another feedback is we probably want to add documentation about this new feature in the Usage section in the readme, to know how to use this new command.

But that's it; the rest looks super good. Thanks again.

**Review by MaferMazu — APPROVED** [review]
This looks good to me. Thanks for this, @bra-i-am ✨

**Pull request merged**